### PR TITLE
Communication - Purchase Search Follow-up PR

### PR DESCRIPTION
**Pull request by jbeauregardb**

### CODE REVIEW

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 1]
```diff
@@ -847,48 +847,53 @@
...
 
     @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PollerFlux<Void, Void> beginPurchaseSearch(String searchId, Duration pollInterval) {
+    public PollerFlux<PhoneNumberSearch, 
+        PhoneNumberSearch> beginPurchaseSearch(String searchId, Duration pollInterval) {
         Objects.requireNonNull(searchId, "'searchId' can not be null.");
-        Objects.requireNonNull(pollInterval, "'pollInterval' can not be null.");
+        
+        if (pollInterval == null) {
+            pollInterval = Duration.ofSeconds(5);
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:855]: Can you make it a static and reuse in all other methods as well?

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 2]
```diff
@@ -847,48 +847,53 @@
...
             purchaseSearchActivationOperation(searchId),
             purchaseSearchPollOperation(searchId),
             (activationResponse, pollingContext) -> Mono.error(new RuntimeException("Cancellation is not supported")),
             purchaseSearchFetchResultOperation());
     }
 
-    private Function<PollingContext<Void>, Mono<Void>> purchaseSearchActivationOperation(String searchId) {
-
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PhoneNumberSearch>> purchaseSearchActivationOperation(String searchId) {
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:866]: nit: do you mind moving Mono<PhoneNumberSearch>> to the line above, will look cleaner :)

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 3]
```diff
@@ -847,48 +847,53 @@
...
         return (pollingContext) -> {
-            Mono<Void> response = purchaseSearch(searchId);
-            return response;
+            Mono<PhoneNumberSearch> res = purchaseSearch(searchId).flatMap(agh -> {
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:868]: Can we use more descriptive parameter names than res, agh?

Comment (ankitarorabit) [reply sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:868]: Also, I find it weird that the argument in flatmap is not used.. is that normal pattern?

**Review by ankitarorabit — CHANGES_REQUESTED** [review]
Added comments

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java (hunk 3 above)
Comment (jbeauregardb) [reply sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:868]: I used agh to test, forgot to remove that!

**Review by ankitarorabit — APPROVED** [review]

**Review by andrewchoi5 — APPROVED** [review]

**Review by ankitarorabit — APPROVED** [review]

**Pull request merged**